### PR TITLE
Basket state design

### PR DESCRIPTION
**Pull request by eoinc134**
State Design for basket, Basket functionality

### CODE REVIEW

**Comment by SeanRyan2000** [comment]
working on jenkins anyways

File: model/Basket/Concessions/AddOns.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
+import six
+from abc import ABCMeta
+from Concessions.Popcorn import Popcorn
+
+@six.add_metaclass(ABCMeta)
+class AddOns(Popcorn):
```
Comment (AndrewMcMahon123) [inline model/Basket/Concessions/AddOns.py:6]: does this class need to extend popcorn? should it not be the other way around?

Comment (AndrewMcMahon123) [reply model/Basket/Concessions/AddOns.py:6]: Like addPopcorn(AddOns)

Comment (eoinc134) [reply model/Basket/Concessions/AddOns.py:6]: No you get popcorn and then addOns is the decorator class for adding more concessions

Comment (AndrewMcMahon123) [reply model/Basket/Concessions/AddOns.py:6]: ohh so you can only get add ons if you order popcorn?

Comment (AndrewMcMahon123) [reply model/Basket/Concessions/AddOns.py:6]: @eoinc134

Comment (eoinc134) [reply model/Basket/Concessions/AddOns.py:6]: ya you start with popcorn and add whatever else you want

**Review by AndrewMcMahon123 — APPROVED** [review]

**Pull request merged**